scripts/install-deps.sh

```diff
@@ -182,6 +206,12 @@ do_install_aduc_packages() {
 
     # Note that clang-tidy requires clang to be installed so that it can find clang headers.
     $SUDO apt-get install --yes "${static_analysis_packages[@]}" || return
+
+    # Check if the directory exists
+    if [ ! -d "$openssl_dir_path" ]; then
```

Comment:
I believe Ubuntu 22.04 uses OpenSSL 3 for the system. Can we instead use newer version of azure-iot-hub-sdk for C that uses OpenSSL 3 when OpenSSL 3 is detected instead of installing OpenSSL 1.1.1 SxS with OpenSSL 3?

Reply:
Yeah I was told that across Microsoft projects including the iot c sdk, they decide not to move to OpenSSL 3.0